Source/Serialization.Json/UnableToInstantiateInstanceOfType.cs

```diff
@@ -0,0 +1,45 @@
+namespace Dolittle.Serialization.Json
+{
+    using System;
+    using System.Runtime.Serialization;
+    
+    /// <summary>
+    /// 
+    /// </summary>
+    [Serializable]
```

Comment:
This is actually a legacy thing for the Binary Serializer when going between multiple AppDomains and the marshalling that was done. It is not needed - nor all the overloads and the protected constructor.

Reply:
Good that we can now drop the Serializable version, never really used it anyway.

On a more general point, I like to create lots of custom exceptions and like it when a custom exception is only thrown from one place.  This tends to mean that I know exactly what is required in the ctor, so should we drop the overloads?  The guidance suggests default ctor, (string), and (string, inner exception) but since we aren't following other parts of the guidance can we just do exactly what we need?

Reply:
I think we should drop them. The .NET guidance is based on very generic approaches. We can make this much clearer in our guidance. There is nothing in the framework or runtime requiring this.